server/config/v2_deprecation.go

```diff
@@ -17,47 +17,50 @@ package config
 type V2DeprecationEnum string
 
 const (
-	// No longer supported in v3.6
+	// V2Depr0NotYet means v2store isn't deprecated yet.
+	// Default in v3.5, and no longer supported in v3.6.
 	V2Depr0NotYet = V2DeprecationEnum("not-yet")
-	// No longer supported in v3.6
-	//
-	// Deprecated: Please use V2Depr0NotYet.
+
+	// Deprecated: to be decommissioned in 3.7. Please use V2Depr0NotYet.
+	// TODO: remove in 3.7
 	//revive:disable-next-line:var-naming
 	V2_DEPR_0_NOT_YET = V2Depr0NotYet
-	// Default in v3.6.  Meaningful v2 state is not allowed.
-	// The V2 files are maintained for v3.5 rollback.
 
-	V2Depr1WriteOnly = V2DeprecationEnum("write-only")
+	// V2Depr1WriteOnly means only writing v2store is allowed.
 	// Default in v3.6.  Meaningful v2 state is not allowed.
 	// The V2 files are maintained for v3.5 rollback.
-	//
-	// Deprecated: Please use V2Depr1WriteOnly.
+	V2Depr1WriteOnly = V2DeprecationEnum("write-only")
+
+	// Deprecated: to be decommissioned in 3.7. Please use V2Depr1WriteOnly.
+	// TODO: remove in 3.7
 	//revive:disable-next-line:var-naming
 	V2_DEPR_1_WRITE_ONLY = V2Depr1WriteOnly
 
-	// V2store is WIPED if found !!!
+	// V2Depr1WriteOnlyDrop means v2store is WIPED if found !!!
+	// Will be default in 3.7.
 	V2Depr1WriteOnlyDrop = V2DeprecationEnum("write-only-drop-data")
-	// V2store is WIPED if found !!!
-	//
-	// Deprecated: Pleae use V2Depr1WriteOnlyDrop.
+
+	// Deprecated: to be decommissioned in 3.7. Pleae use V2Depr1WriteOnlyDrop.
+	// TODO: remove in 3.7
 	//revive:disable-next-line:var-naming
 	V2_DEPR_1_WRITE_ONLY_DROP = V2Depr1WriteOnlyDrop
 
-	// V2store is neither written nor read. Usage of this configuration is blocking
+	// V2Depr2Gone means v2store is completely gone. The v2store is
+	// neither written nor read. Anything related to v2store will be
+	// cleaned up in v3.8. Usage of this configuration is blocking
 	// ability to rollback to etcd v3.5.
 	V2Depr2Gone = V2DeprecationEnum("gone")
-	// V2store is neither written nor read. Usage of this configuration is blocking
-	// ability to rollback to etcd v3.5.
-	//
-	// Deprecated: Please use V2Depr2Gone
+
+	// Deprecated: to be decommissioned in 3.7. Please use V2Depr2Gone.
+	// TODO: remove in 3.7
 	//revive:disable-next-line:var-naming
 	V2_DEPR_2_GONE = V2Depr2Gone
 
-	// Default deprecation level.
+	// V2DeprDefault is the default deprecation level.
 	V2DeprDefault = V2Depr1WriteOnly
```

Comment:
Is `V2Depr1WriteOnly` the correct v2 default config for v3.6? From above 
```
	// V2Depr1WriteOnly means only writing v2store is allowed.
	// Default in v3.6.  Meaningful v2 state is not allowed.
	// The V2 files are maintained for v3.5 rollback.
```

However we plan to panic on any v2 state. Should we reconcile this?

Reply:
> Is `V2Depr1WriteOnly` the correct v2 default config for v3.6?

YES.



> However we plan to panic on any v2 state.

In 3.6, we only allow membership data in v2state. We will keep it as it's in 3.6. In 3.7, the default level will be `V2Depr1WriteOnlyDrop`, and we will remove any v2 snapshot files if found.

https://github.com/etcd-io/etcd/blob/fbac2a064c381d6f8c74c4d9d01fde816e200eef/server/etcdserver/bootstrap.go#L407-L409

Reply:
I thought `V2DeprecationEnum` was about key-value data stored in v2 store, not about membership data. 

To clarify, enums are not accurate to the current state, as the plan has changed in https://github.com/etcd-io/etcd/issues/12913#issuecomment-1531941532.